api/client.go

```diff
@@ -457,3 +458,35 @@ func HandleConnError(err error) error {
 
 	return nil
 }
+
+func ServerCompatible(ctx context.Context, client Client) error {
+	srv, err := client.GetServerVersion(ctx)
+	if err != nil {
+		logging.L.Debug().Err(err).Msg("check server version")
+		return nil // result inconclusive
+	}
+	ahead, err := versionAhead(client.Version, srv.Version)
+	if err != nil {
+		logging.L.Debug().Err(err).Msg("failed to compare client version to server")
+		return nil // result inconclusive
+	}
+	if ahead {
+		return fmt.Errorf("client version (%s) is ahead of server version (%s), please upgrade your server", apiVersion, srv.Version)
```

Comment:
Since it's kinda unrealistic to tell the user to upgrade their server, we should also include a note which says that they can downgrade their client, along with instructions on how to do that.

Reply:
Added the link:
```
Error: client version (0.18.1) is ahead of server version (99.99.99999), download the CLI version that matches your server, reference https://infrahq.com/docs/reference/self-hosting#cli-versions
```

Let me know what you think of the wording.

Reply:
@BruceMacD can you edit the link to https://infrahq.com/docs/messages/cli-versions as a "short link" and I'll make sure the redirect goes to above

Reply:
^ updated